### PR TITLE
MINOR: Remove accidental unnecessary code in src

### PR DESCRIPTION
**Pull request by clolov**
I ran IntelliJ IDEA's Code Inspection and corrected occurrences of the following:
* https://www.jetbrains.com/help/idea/list-of-java-inspections.html#inheritance-issues (method is identical to its super method)
* https://www.jetbrains.com/help/idea/list-of-java-inspections.html#imports (unused import)

It is safe to remove said methods because they are already defined upward in the hierarchy chain.

### CODE REVIEW

**Comment by clolov** [comment]
Hello @mimaison, would you have the time to review this?

**Comment by mimaison** [comment]
I checked the clients and connect changes and they look good. I think some of the streams changes could be debatable (not sure if they want to keep some of the comments) so I'll let someone working on streams review them. @ableegoldman @guozhangwang can you take a look?

**Comment by mimaison** [comment]
Thanks for the updates!

There's a spotbugs failure:
```
Bug type EQ_DOESNT_OVERRIDE_EQUALS
In class org.apache.kafka.connect.runtime.InternalSinkRecord

EQ_DOESNT_OVERRIDE_EQUALS: Class doesn't override equals in superclass

This class extends a class that defines an equals method and adds fields, but doesn't define an equals method itself. Thus, equality on instances of this class will ignore the identity of the subclass and the added fields. Be sure this is what is intended, and that you don't need to override the equals method. Even if you don't need to override the equals method, consider overriding it anyway to document the fact that the equals method for the subclass just return the result of invoking super.equals(o). 
```

**Comment by clolov** [comment]
Great, thank you, I will address this!

**Review by mimaison — APPROVED** [review]
Thanks, LGTM

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SourceConnectorConfig.java [hunk 1]
```diff
@@ -117,10 +117,6 @@ private static class EnrichedSourceConnectorConfig extends ConnectorConfig {
             super(plugins, configDef, props);
         }
 
-        @Override
-        public Object get(String key) {
-            return super.get(key);
```
Comment (chia7712) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SourceConnectorConfig.java:122]: It seems `EnrichedSourceConnectorConfig` is no longer needed. If so, I can file a follow-up patch to clean it up

**Pull request merged**